frontend/packages/db-structure/src/parser/drizzle/index.test.ts

```diff
@@ -0,0 +1,757 @@
+    })
+
+    it('many-to-many relationship with junction table', async () => {
+      const { value } = await processor(`
+        import { pgTable, serial, integer, varchar, primaryKey } from 'drizzle-orm/pg-core';
+        import { relations } from 'drizzle-orm';
+
+        export const users = pgTable('users', {
+          id: serial('id').primaryKey(),
+          name: varchar('name', { length: 255 }),
+        });
+
+        export const tags = pgTable('tags', {
+          id: serial('id').primaryKey(),
+          name: varchar('name', { length: 100 }).unique(),
+        });
+
+        export const userTags = pgTable('user_tags', {
+          userId: integer('user_id').references(() => users.id).notNull(),
+          tagId: integer('tag_id').references(() => tags.id).notNull(),
+        }, (table) => ({
+          pk: primaryKey({ columns: [table.userId, table.tagId] }),
+        }));
+
+        export const usersRelations = relations(users, ({ many }) => ({
+          userTags: many(userTags),
+        }));
+
+        export const tagsRelations = relations(tags, ({ many }) => ({
+          userTags: many(userTags),
+        }));
+
+        export const userTagsRelations = relations(userTags, ({ one }) => ({
+          user: one(users, {
+            fields: [userTags.userId],
+            references: [users.id],
+          }),
+          tag: one(tags, {
+            fields: [userTags.tagId],
+            references: [tags.id],
+          }),
+        }));
+      `)
+
+      // Verify junction table exists
+      expect(value.tables).toHaveProperty('user_tags')
+
+      // Verify foreign key constraints in junction table
+      expect(value.tables['user_tags']?.constraints).toHaveProperty(
+        'user_tags_user_id_users_id_fk',
+      )
+      expect(
+        value.tables['user_tags']?.constraints['user_tags_user_id_users_id_fk'],
+      ).toEqual({
+        type: 'FOREIGN KEY',
+        name: 'user_tags_user_id_users_id_fk',
+        columnName: 'user_id',
+        targetTableName: 'users',
+        targetColumnName: 'id',
+        updateConstraint: 'NO_ACTION',
+        deleteConstraint: 'NO_ACTION',
+      })
+
+      expect(value.tables['user_tags']?.constraints).toHaveProperty(
+        'user_tags_tag_id_tags_id_fk',
+      )
+      expect(
+        value.tables['user_tags']?.constraints['user_tags_tag_id_tags_id_fk'],
+      ).toEqual({
+        type: 'FOREIGN KEY',
+        name: 'user_tags_tag_id_tags_id_fk',
+        columnName: 'tag_id',
+        targetTableName: 'tags',
+        targetColumnName: 'id',
+        updateConstraint: 'NO_ACTION',
+        deleteConstraint: 'NO_ACTION',
+      })
+
+      // Verify composite primary key
+      expect(value.tables['user_tags']?.constraints).toHaveProperty(
+        'user_tags_pkey',
+      )
+      expect(value.tables['user_tags']?.constraints['user_tags_pkey']).toEqual({
+        type: 'PRIMARY KEY',
+        name: 'user_tags_pkey',
+        columnName: 'user_id,tag_id',
```

Comment:
The composite primary key is not expressed now in db-structure, but it looks like it should be fixed to match the following:

https://github.com/liam-hq/liam/issues/2260